lib/hanami/assets/helpers.rb

```diff
@@ -186,9 +240,25 @@ def javascript(*sources)
       #   <%= stylesheet 'application' %>
       #
       #   # <link href="https://assets.bookshelf.org/assets/application-28a6b886de2372ee3922fcaf3f78f2d8.css" type="text/css" rel="stylesheet">
-      def stylesheet(*sources)
+      def stylesheet(*sources, **options)
         _safe_tags(*sources) do |source|
-          html.link(href: _typed_asset_path(source, STYLESHEET_EXT), type: STYLESHEET_MIME_TYPE, rel: STYLESHEET_REL).to_s
+          tag_options = options.merge({
```

Comment:
@jodosha I agree that 1 is the right path, and developer should be responsible for their actions. :smile: 

I've already made changes (uncommitted) to allow the full scope of attributes to be overloaded. What I could do is add a simple sanity check where multiple sources are provided along with asset specific attributes, such as in your example:

```erb
<%= javascript 'jquery', 'application', id: 'my script' %>
``` 

The above could log a warning.